### PR TITLE
Fix potential deadlock in DB backed resource group manager

### PR DESCRIPTION
**Pull request by cberner**
The memory pool listener acquires locks in the order
generalPoolMemoryFraction -> root group
However, configureGroup() acquires them in the order root group ->
generalPoolMemoryFraction

### CODE REVIEW

**Comment by cberner** [comment]
Fixes: https://github.com/prestodb/presto/issues/6755

File: presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/AbstractResourceConfigurationManager.java [hunk 1]
```diff
@@ -98,13 +98,17 @@ private void validateSelectors(List<ResourceGroupSpec> groups, List<ResourceGrou
     protected AbstractResourceConfigurationManager(ClusterMemoryPoolManager memoryPoolManager)
     {
         memoryPoolManager.addChangeListener(new MemoryPoolId("general"), poolInfo -> {
+            Map<ResourceGroup, DataSize> memoryLimits = new HashMap<>();
```
Comment (nezihyigitbasi) [inline presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/AbstractResourceConfigurationManager.java]: maybe we can add a comment here of why we put the limits in a map and then apply.

**Comment by nezihyigitbasi** [comment]
@cberner thanks for the fix!

File: presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/AbstractResourceConfigurationManager.java (hunk 1 above)
Comment (cberner) [inline presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/AbstractResourceConfigurationManager.java]: Good idea, added a comment

**Pull request merged**